.husky/check-versions.js

```diff
@@ -0,0 +1,23 @@
+import fs from 'node:fs';
+
+// Read package.json
+const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+
+// Extract versions
+const devServerDependencyVersion = packageJson.dependencies['@lwc/lwc-dev-server'];
+const devServerTargetVersion = packageJson.apiVersionMetadata?.target?.matchingDevServerVersion;
+
+if (!devServerDependencyVersion || !devServerTargetVersion) {
+  console.error('Error: missing @lwc/lwc-dev-server or matchingDevServerVersion');
+  process.exit(1); // Fail the check
+}
+
+// Compare versions
+if (devServerDependencyVersion === devServerTargetVersion) {
+  process.exit(0); // Pass the check
+} else {
+  console.error(
+    `Error: Versions do not match. @lwc/lwc-dev-server: ${devServerDependencyVersion}, matchingDevServerVersion: ${devServerTargetVersion}`
```

Comment:
I would make this error message more reflective of the process request being conveyed here:
```suggestion
    `Error: @lwc/lwc-dev-server versions do not match between 'dependencies' and 'apiVersionMetadata'. Expected ${devServerDependencyVersion} in apiVersionMetadata > target > matchingDevServerVersion. Got ${devServerTargetVersion} instead. When updating the @lwc/lwc-dev-server dependency, you must ensure that it's compatible with the supported API version in this branch, then update apiVersionMetadata > target > matchingDevServerVersion to match, in order to "sign off" on this dependency change.`
```